qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/QuteSupportForTemplate.java

```diff
@@ -328,21 +338,107 @@ private static IJavaProject getJavaProjectFromProjectUri(String projectName) {
+	 * @param utils       the JDT utilities.
+	 * 
+	 * @return the Java project of the given resource Uri (file or project uri) and
+	 *         null otherwise.
+	 */
+	private static IJavaProject findJavaProject(String resourceUri, IJDTUtils utils) {
+		IProject project = findProject(resourceUri, utils);
+		if (project == null) {
 			// The uri doesn't belong to an Eclipse project
 			return null;
 		}
 		// The uri belong to an Eclipse project
-		if (!(JavaProject.hasJavaNature(file.getProject()))) {
+		return getJavaProject(project);
+	}
+
+	/**
+	 * Returns the project of the given resource Uri (file or project uri) and null
+	 * otherwise.
+	 * 
+	 * @param resourceUri the resource Uri (file or project uri).
+	 * @param utils       the JDT utilities.
+	 * 
+	 * @return the project of the given resource Uri (file or project uri) and null
+	 *         otherwise.
+	 */
+	private static IProject findProject(String resourceUri, IJDTUtils utils) {
+		String fileUri = resourceUri.replace("vscode-notebook-cell", "file");
+		IFile file = utils.findFile(fileUri);
+		if (file != null) {
+			// The resource Uri is a file which belong to an Eclipse project
+			return file.getProject();
+		}
+
+		// Try to get the project
+		URI projectUri = toURI(resourceUri);
+		if (!URIUtil.isFileURI(projectUri)) {
+			// ex : untitled:untitled
+			return null;
+		}
+		Path projectPath = Paths.get(projectUri);
+		for (IProject currentProject : ResourcesPlugin.getWorkspace().getRoot().getProjects()) {
+			Path currentProjectPath = Paths.get(currentProject.getLocationURI());
+			if (projectPath.equals(currentProjectPath)) {
+				// check if it is a Qute project
+				if (!isQuteProject(currentProject)) {
+					return null;
+				}
+				return currentProject;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Returns true if the given project is a Java Qute project and false otherwise.
+	 * 
+	 * @param project the project.
+	 * 
+	 * @return true if the given project is a Java Qute project and false otherwise.
+	 */
+	private static boolean isQuteProject(IProject project) {
+		IJavaProject javaProject = getJavaProject(project);
+		if (javaProject == null) {
+			return false;
+		}
+		return JDTTypeUtils.findType(javaProject, QuteJavaConstants.TEMPLATE_CLASS) != null;
+	}
+
+	/**
+	 * Returns the Java project of the given project and null otherwise.
+	 * 
+	 * @param project the project
+	 * 
+	 * @return the Java project of the given project and null otherwise.
+	 */
+	private static IJavaProject getJavaProject(IProject project) {
+		if (!(JavaProject.hasJavaNature(project))) {
 			// The uri doesn't belong to a Java project
 			return null;
 		}
+		return JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+	}
 
-		String projectName = file.getProject().getName();
-		return JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(projectName);
+	private static URI toURI(String uriString) {

```

Comment:
Could we move this to a utils class? Seems like it would be useful elsewhere in the future